### PR TITLE
fix: #8 window is undefined error

### PR DESCRIPTION
**Pull request by alias-rahil**
Closes https://github.com/protocolbuffers/protobuf-javascript/issues/8. Fixes the issue by explicitly checking if window is defined.

> NOTE: This PR is similar to the old repo PR https://github.com/protocolbuffers/protobuf/pull/9156 by @MarnixBouhuis

### CODE REVIEW

**Comment by MarnixBouhuis** [comment]
Looks like https://github.com/protocolbuffers/protobuf/pull/9156/ didn't got copied over when the generator was moved to this repository

File: generator/js_generator.cc [hunk 1]
```diff
@@ -3637,8 +3637,13 @@ void Generator::GenerateFile(const GeneratorOptions& options,
       // may be blocked by things like CSP.
       //   Function('') is almost the same as eval('')
       printer->Print(
-          "var global = (function() { return this || window || global || self "
-          "|| Function('return this')(); }).call(null);\n\n");
+          "var global = (function() {\n"
+          "  if (this) { return this; }\n"
+          "  if (typeof window !== 'undefined') { return window; }\n"
```
Comment (lukesandberg) [inline generator/js_generator.cc:3642]: we should add globalThis to the set of symbols to check for and check for that one first

File: generator/js_generator.cc [hunk 2]
```diff
@@ -3637,8 +3637,13 @@ void Generator::GenerateFile(const GeneratorOptions& options,
       // may be blocked by things like CSP.
       //   Function('') is almost the same as eval('')
       printer->Print(
-          "var global = (function() { return this || window || global || self "
-          "|| Function('return this')(); }).call(null);\n\n");
+          "var global = (function() {\n"
```
Comment (lukesandberg) [inline generator/js_generator.cc:3640]: move the checks for globalThis, window, self, global outside of the function, so in the common cases where they exist we can avoid constructing and invoking the function objects

File: generator/js_generator.cc (hunk 1 above)
Comment (alias-rahil) [reply generator/js_generator.cc:3644]: Done!

File: generator/js_generator.cc (hunk 2 above)
Comment (alias-rahil) [reply generator/js_generator.cc]: Moved the global symbols checks outside https://github.com/protocolbuffers/protobuf-javascript/pull/106/commits/d3967d0b53a22dc90b362b49e7d5e0e286afd73c. Let me know if I misunderstood you or you'd like me to revert the commit.

Comment (lukesandberg) [reply generator/js_generator.cc]: yep, you got it.

Can you inline all the new variables into one big expression so we can avoid polluting the global scope.

var global = (typeof globalThis !== 'undefined' && globalThis) || (.....)

**Review by lukesandberg — CHANGES_REQUESTED** [review]

File: generator/js_generator.cc (hunk 2 above)
Comment (alias-rahil) [reply generator/js_generator.cc]: Done!

**Review by lukesandberg — APPROVED** [review]

**Pull request merged**